src/build.js

```diff
@@ -9,15 +9,27 @@ import { getMarkdownFiles } from './getFiles.js'
 import { sourceAndInstallButton } from './github.js';
 import { loadMetadata } from './userscriptMetadata.js';
 
+/**
+ * Returns the formatted name of the userscript for use in documentation.
+ * @type {import('./types/UserscriptBuildOptions.js').UserscriptNameFormatter}
+ */
+function defaultNameFormatter({metadata}) {
+	return metadata.name;
+}
+
 export async function build({
 	bookmarkletBasePath = false,
-	userscriptBasePath = 'src/userscripts',
+	userscripts = {
+		sourcePath: 'src/userscripts',
+		destPath: 'dist',
+		nameFormatter: defaultNameFormatter,
```

Comment:
There is one problem with this nested parameter: The fallback values only work for the first level of the parameter object which will be destructured and has defaults specified with `=`.
Unfortunately this syntax can't be used for the inner object, so the following won't work:
```js
export async function build({
	bookmarkletBasePath = false,
	userscriptBasePath = 'src/userscripts',
	userscripts = {
		sourcePath = 'src/userscripts',
		destPath = 'dist',
		nameFormatter = defaultNameFormatter,
	},
	...
} = {})
```
So we either have to provide defaults in the function body or discard the approach to use nested parameters.

Reply:
How about we discard the nested parameter approach and go back to prefixing the variable names with `userscript` instead? I think defining defaults inside the function it too opaque and reduces maintainability. Also the main benefit of object destructuring for me is that we can define defaults in the function header and the caller only needs to specify the arguments they want to change.

So something like:

```js
export async function build({
	userscriptSourcePath = 'src/userscripts',
	userscriptDestPath = 'dist',
	userscriptNameFormatter = defaultNameFormatter,
	bookmarkletSourcePath = 'src/bookmarklets',
	docSourcePath = 'doc',
	readmePath = 'README.md',
	debug = false,
} = {})
```

Reply:
Yeah, that was also my thought after testing the behaviour of nested parameters, good to have it confirmed. It's better to give up the idea of passing an entire parameter object down to an underlying function, having to reassign multiple parameters to `buildUserscripts` is still better than having to deal with fallback values in the function body again.